pkg/handler/enqueue_annotation.go

```diff
@@ -0,0 +1,173 @@
+//	      os.Exit(1)
+//    }
+// }
+// ...
+// With the annotations:
+//
+// ...
+// annotations:
+//    sigs.k8s.io/primary-resource-name: "my-namespace/my-replicaset"
+//    sigs.k8s.io/primary-resource-type: "ReplicaSet.apps"
+// ...
+type EnqueueRequestForAnnotation struct {
+	// It is used to verify that the primary resource is the primary resource to use. It can be any arbitrary value.
+	// However, it would be recommended to fill it with `<Kind>.<group>` from the primary resource to make it easy
+	// to tell which specific resource is interested in.
+	Type string
+}
+
+// Create implements EventHandler
+func (e *EnqueueRequestForAnnotation) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// Update implements EventHandler
+func (e *EnqueueRequestForAnnotation) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.MetaOld); ok {
+		q.Add(req)
+	} else if ok, req := e.getAnnotationRequests(evt.MetaNew); ok {
+		q.Add(req)
+	}
+}
+
+// Delete implements EventHandler
+func (e *EnqueueRequestForAnnotation) Delete(evt event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// Generic implements EventHandler
+func (e *EnqueueRequestForAnnotation) Generic(evt event.GenericEvent, q workqueue.RateLimitingInterface) {
+	if ok, req := e.getAnnotationRequests(evt.Meta); ok {
+		q.Add(req)
+	}
+}
+
+// getAnnotationRequests will check if the object has the annotations for the watch handler and requeue
+func (e *EnqueueRequestForAnnotation) getAnnotationRequests(object metav1.Object) (bool, reconcile.Request) {
+	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type {
+		if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type {
+			namespacedNameString, ok := object.GetAnnotations()[NamespacedNameAnnotation]
+			if !ok {
+				log.Info("Unable to find the annotation for handle watch annotation",
+					"resource", object, "annotation", NamespacedNameAnnotation)
+			}
+			if len(namespacedNameString) < 1 {
+				return false, reconcile.Request{}
+			}
+			return true, reconcile.Request{NamespacedName: parseNamespacedName(namespacedNameString)}
+		}
+		return false, reconcile.Request{}
+	}
+	return false, reconcile.Request{}
+}
+
+// parseNamespacedName will parse the value informed in the NamespacedNameAnnotation and return types.NamespacedName
+// with. Note that if just one value is informed, then it will be set as the name.
+func parseNamespacedName(namespacedNameString string) types.NamespacedName {
+	values := strings.Split(namespacedNameString, "/")
+	if len(values) == 1 {
+		return types.NamespacedName{
+			Name:      values[0],
+			Namespace: "",
+		}
+	}
+	if len(values) >= 2 {
+		return types.NamespacedName{
+			Name:      values[1],
```

Comment:
I don't remember if "/" is allowed in name, but a safer way to handle this is to use `strings.SplitN(namespacedNameString, "/", 2)`.